### ISSUE TITLE
*.From*(decimal) overload is not present.

### ISSUE THREAD

**Issue by aignas** [angularsen/UnitsNet#282]
Hello,

I noticed a small discrepancy in the API. Imagine that I have a `decimal`, which may be `nullable`, but that is not the main thing. In order to convert it to a Length type, for example, I have one of the following options:

```
decimal length = 3.14;
Length length = From.Meters((double)length);
```

or using the extension methods:

```
decimal length = 3.14;
Length length = length.Meters();
```

Am I missing something or is it indeed the case that the overloads for decimals are missing? Would you be willing to have this in the library?

**Comment by anjdreas** [comment]
Hi, yes decimal overloads are missing. In retrospect, if I had written this today I would probably have picked decimal as the backing type to begin with, instead of double. But I'm all for adding `From` method overloads at least. It would still be represented by double internally and you would still get double in all the conversion properties, like `double m = myLength.Meters`, but it would be more convenient for code that already works with decimal. Also, I'm not sure how `double` to `decimal` assignment works and whether you need/want to use `decimal d = Convert.ToDecimal(myLength.Meters)` to retain more precision.

A couple of options:
1) Generate `From` method overloads taking `decimal` and `decimal?` instead of `double` and `double?`. 
2) Generate a duplicate of all quantity types, using `decimal` as internal representation and either call it `LengthD` vs `Length` or put it in a different namespace `UnitsNet.Decimal.Length` vs `UnitsNet.Length`. We could also consider releasing these new types in a separate nuget to avoid doubling the existing nuget size, such as a new nuget `UnitsNet.Decimal`.

Which option would you consider the best for your case?

**Comment by aignas** [comment]
Hi, thanks for the reply. I think that adding the overloads would be a nice addition. In my case I need to consume decimals and convert everything into `Length`s and I am not terribly concerned with the backing type, however, I appreciate that there may be a case where this distinction may be important.

If it was up to me, I would do the following:
#. Add extra overloads taking `decimal` and `decimal?` and remove the `double` and `double?` overloads only if necessary.
#. Add an extra NuGet called `UnitsNet.Decimal` if we need the `decimal` as the backing type. However, I see a challenge for existing code bases where converting all `using` statements to a different `namespace` may pose a challenge, but at the same time I am not aware if `NuGet` provides a mechanism to define conflicting packages.

As the second step may require much more effort, I would open a separate issue for this. I would be quite happy to say that the issue is solved for me once the option 1 is implemented. :)

**Comment by eriove** [comment]
(Me and @aignas are working in the same team.)

I agree with @aignas that not removing the `double` and `double?` overloads is preferable.

I'm personally afraid that switching to `decimal` internally would have a large performance impact in our project. One NuGet package with `decimal` and one with `double` would be the preferred solution for us.

The risk is that this would make the library less approachable and harder to get started with.

**Comment by eriove** [comment]
I started working on `decimal` and had to add `int` and `long` too for existing code not to break. It all took much longer than expected. Will continue tomorrow.

**Comment by angularsen** [comment]
Fixed in #298
